### DISCUSSION TITLE
Board Configuration

### DISCUSSION THREAD

**Discussion by danielerpl** [DIY-Flow-Bench/DIY-Flow-Bench#149] [Code Development]
Hi, I ma trying to use my ESP32 Board 
it is a ESP32-WROOM-32 pretty similar/equal to ESP32DUINO 

I created the configuration in the pins.h but I am still receiving this error. 
Currently I don't have nothing attached to the board. It's nude, could be this the problem ? 
I am using VSCode with the last release as described in the error 

I am sure I need to do something else with the configuration 

DIY Flow Bench 
Version: V.2.0-RC.6 
Build: 23041301 
Initialising Hardware 
E (28) gpio: GPIO can only be used as input mode
[    44][E][esp32-hal-gpio.c:130] __pinMode(): GPIO config failed
Scanning for I2C devices...
Found 0 device(s). 
Initialising ADS1115 
[  1077][E][Wire.cpp:513] requestFrom(): i2cRead returned Error 263
ADS1115 Connection failed

**Comment by DeeEmm** [comment]
It appears to be throwing an error on pin 28. (you also do not have BME or ADC connected to I2C but I guess you already know this :) )

Can you share the pin map that you created and also the manufacturers pin layout for the board as there are several different manufacturers with several different layouts.

e.g.

![esp32-pinout-chip-ESP-WROOM-32](https://github.com/DeeEmm/DIY-Flow-Bench/assets/3038710/fda3e59a-508b-4394-abbf-c49b9000c1cb)

**Comment by DeeEmm** [comment]
Any luck with this?

**Comment by danielerpl** [comment]
Hi Mick

sorry for the late reply, I am so busy with my develop job and I completely
forgot to check this email.

So, the board i am usi is exactly this one

[image: image.png]

Il giorno ven 12 mag 2023 alle ore 23:35 Mick Percy <
***@***.***> ha scritto:

**Reply by danielerpl** [reply]
![714PSaVZQtL _SL1500_](https://github.com/DeeEmm/DIY-Flow-Bench/assets/14088795/03a2169a-cd6d-4069-a24b-6fcedc0cd2ce)

**Reply by DeeEmm** [reply]
Can you also share the pin map that you created

**Comment by danielerpl** [comment]
/***********************************************************
 * ESP32-WROOM-32
 *
 ***/
#ifdef ESP32_WROOM_32

    #define BOARD_TYPE              "ESP32_WROOM_32"

    // Define Physical Pins
    
    // VAC CONTROL
    #define VAC_SPEED_PIN               5                       // Built in DAC1 - used for speed reference for VFD (0-3v)
    #define VAC_BLEED_VALVE_PIN         26                      // Built in DAC2 - used for bleed valve control
    
    #define VAC_BANK_1_PIN              13                      // vac motor(s) on/off
    #define VAC_BANK_2_PIN              12                      // Provision for 2 stage Vac motor control
    #define VAC_BANK_3_PIN              14                      // Provision for 3 stage Vac motor control
    
    //STEPPER MOTOR CONTROLLER                   
    #define AVO_STEP_PIN                19                      // NOTE 19 Also used by SD Card
    #define AVO_DIR_PIN                 23                      // NOTE 23 Also used by SD Card
                  
    #define FLOW_VALVE_STEP_PIN         27
    #define FLOW_VALVE_DIR_PIN          35

    #define VCC_3V3_PIN                 99                      
    #define VCC_5V_PIN                  2                      // 10k-10k divider across 5v supply
    
    // SENSORS
    #define SPEED_SENSOR_PIN            5                      // NOTE Also used by SD Card  // turbine / rotor speed for turbo / blower flow bench 

     // SWIRL ENCODER
    #define SWIRL_ENCODER_PIN_A         99
    #define SWIRL_ENCODER_PIN_B         99

   // ORIFICE DETECTION                                                                                    
    #define ORIFICE_BCD_BIT1_PIN        34                                       
    #define ORIFICE_BCD_BIT2_PIN        36                                           
    #define ORIFICE_BCD_BIT3_PIN        39                         

    // NOTE: these inputs are handled by ADC
    #define MAF_PIN                     99                     // NOTE: I2C ADC is used instead
    #define REF_PRESSURE_PIN            99                     // NOTE: I2C ADC is used instead
    #define DIFF_PRESSURE_PIN           99                     // NOTE: I2C ADC is used instead
    #define PITOT_PIN                   99                     // NOTE: I2C ADC is used instead
    
    // NOTE: These inputs are handled by BME280
    #define TEMPERATURE_PIN             99                     // NOTE: I2C BME280 used
    #define REF_BARO_PIN                99                     // NOTE: I2C BME280 used
    #define HUMIDITY_PIN                99                     // NOTE: I2C BME280 used                     
    
    // COMMS    
    #define SERIAL0_TX_PIN              1                       // API
    #define SERIAL0_RX_PIN              3                       // API
    #define SERIAL2_TX_PIN              16                      // GAUGE PROTOCOL CLOCK
    #define SERIAL2_RX_PIN              17                      // GAUGE PROTOCOL DATA
    #define SDA_PIN                     21                      // BME280 etc
    #define SCL_PIN                     22                      // BME280 etc

    // SD Card
    #define SD_CS_PIN                   5
    #define SD_MOSI_PIN                 23                 
    #define SD_MISO_PIN                 19             
    #define SD_SCK_PIN                  18                   

    
    // NOTE: Pins 15/32/33 are only available on Wemos D1 R32
    // #define WEMOS_SPARE_PIN_1           15                      // NOTE: Pin 15 cannot be used for ADC (Analog in)
    // #define WEMOS_SPARE_PIN_2           32
    // #define WEMOS_SPARE_PIN_3           33                    


#endif


I thins it's completely wrong this mapping and I have to check it again

**Reply by DeeEmm** [reply]
OK. So pins 34-39 can only be used as inputs. I note that you have the FLOW_VALVE_DIR_PIN set to 35 I think that this might be the cause of the error you are seeing.

There is a full description of pin types and assignments in the Expressif documentation. This is a good place to start to check for issues like this...

https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/gpio.html

Let me know how you get on 

/DM

**Comment by danielerpl** [comment]
Right, fixed the error about GPIO I moved the pin 35 to the 32, this is a corre I/O GPIO 
So, it's remaining this error 

DIY Flow Bench 
Version: V.2.0-RC.6 
Build: 23041301 
Initialising Hardware 
Scanning for I2C devices...
Found 0 device(s). 
Initialising ADS1115 
[  1062][E][Wire.cpp:513] requestFrom(): i2cRead returned Error 263
ADS1115 Connection failed


At the moment I don't have any sensor connected to the board. Should be this the problem ?

**Reply by DeeEmm** [reply]
Yes. wire.cpp is used to communicate with the I2C devices. This error should clear when you install the BME and ADS devices. You should also be able to disable these in the configuration if you want to used a fixed value for temp / baro / humidity and assign pin values for the pressure and MAF sensors. However whilst this functionality is in the code, it is really just for testing.

**Reply by danielerpl** [reply]
Ok, I will try to use fixed value at the moment. I will let you know when I have a little bit time to test it ;)

**Comment by DeeEmm** [comment]
I'm going to mark this as resolved. 

I've also added your pin mapping to pins.h so that anyone else who wants to use the same style of ESP32 can use it. If you update it or  change it when you commission your setup please feedback the changes to me so that I can update pins.h

**Reply by danielerpl** [reply]
Amazing!! thanks for the supporto on it

**Comment by danielerpl** [comment]
@DeeEmm if I would use this board in the picture, can you give me info about to wire this one with the pub board ? 
Which is the required pin to have connected ? 
Many thanks 

![image](https://github.com/DeeEmm/DIY-Flow-Bench/assets/14088795/7e84d27b-ffb3-43fb-9d2e-c8cc63f97b49)

**Reply by DeeEmm** [reply]
You need to compare the pin mapping for this board to the pin mapping for an existing board and create a new pin mapping in pins.h for the new board.

It would pay to do this before purchasing the board as some pins may not be brought out to the terminals or may be functionally different. In this case it may be necessary to re-assign pins. 

Generally most ESP32 BOBs share similar pin functionality

**Reply by danielerpl** [reply]
@DeeEmm  solved by getting you same board! I am now in your same situation 

Now I plugged each pin from the WeMos to the pcb, but I have this error with the last release. 

DIY Flow Bench 
Version: V.2.0-RC.6 
Build: 23041301 
Initialising Hardware 
Scanning for I2C devices...
Found 0 device(s). 
Initialising ADS1115 
[128025][E][Wire.cpp:513] requestFrom(): i2cRead returned Error 263
ADS1115 Connection failed


I make few code fixing in the class AsyncTCP.cpp 
there are some errors in 2 functions about the 'if' flow. Let me know if you want see the changes, for me was making an error during the building of the code

**Reply by danielerpl** [reply]
by disabling the ADC_IS_ENABLED    
I have this error, It's like there is not connection between the esp32 and the pcb 

DIY Flow Bench 
Version: V.2.0-RC.6 
Build: 23041301 
Initialising Hardware 
Scanning for I2C devices...
Found 0 device(s). 
Hardware Initialised 
Initialising Sensors 
Initialising BME280 
[128038][E][Wire.cpp:513] requestFrom(): i2cRead returned Error 263
[130040][E][Wire.cpp:513] requestFrom(): i2cRead returned Error 263
BME sensor did not respond. 
Please check wiring and I2C address
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.

**Reply by DeeEmm** [reply]
It looks like it might be an issue with the BME or ADS1115 addressing. Depending on the manufacturer you may find that they have different default addresses to those used in the code.  I would check the addresses for the devices you have and compare them to those used in the code.

**Reply by danielerpl** [reply]
@DeeEmm  I used this one for the ADS11115 
https://www.az-delivery.de/en/products/analog-digitalwandler-ads1115-mit-i2c-interface
and this one for the BME 
https://www.az-delivery.de/en/products/gy-bme280

I tried to make a single easy script just to test the BME but nothing is arriving in the WeMos board

**Comment by danielerpl** [comment]
@DeeEmm I made a test today
I connected a LED with a 1k resistor from 5V to SDA and SCL to the led negative ( - ) pin, anode to 5V. In this scenario the ADS it's correctly initialised. 
But, when I connect the WeMos board with the SCL and SDA of the PCB is not inizialized the ADS. 😩
It's like there is an issue on the pcb board between the ADS and the SDA SCL pin. How can I test it ? any suggestion ? 

I used the code below to test the ADS board 

`#include <Adafruit_ADS1X15.h>

// Adafruit_ADS1115 ads;  /* Use this for the 16-bit version */
Adafruit_ADS1015 ads;     /* Use this for the 12-bit version */

void setup(void)
{
  Serial.begin(115200);
  Serial.println("Hello!");

  Serial.println("Getting differential reading from AIN0 (P) and AIN1 (N)");
  Serial.println("ADC Range: +/- 6.144V (1 bit = 3mV/ADS1015, 0.1875mV/ADS1115)");

  // The ADC input range (or gain) can be changed via the following
  // functions, but be careful never to exceed VDD +0.3V max, or to
  // exceed the upper and lower limits if you adjust the input range!
  // Setting these values incorrectly may destroy your ADC!
  //                                                                ADS1015  ADS1115
  //                                                                -------  -------
  // ads.setGain(GAIN_TWOTHIRDS);  // 2/3x gain +/- 6.144V  1 bit = 3mV      0.1875mV (default)
  // ads.setGain(GAIN_ONE);        // 1x gain   +/- 4.096V  1 bit = 2mV      0.125mV
  // ads.setGain(GAIN_TWO);        // 2x gain   +/- 2.048V  1 bit = 1mV      0.0625mV
  // ads.setGain(GAIN_FOUR);       // 4x gain   +/- 1.024V  1 bit = 0.5mV    0.03125mV
  // ads.setGain(GAIN_EIGHT);      // 8x gain   +/- 0.512V  1 bit = 0.25mV   0.015625mV
  // ads.setGain(GAIN_SIXTEEN);    // 16x gain  +/- 0.256V  1 bit = 0.125mV  0.0078125mV

  if (!ads.begin()) {
    Serial.println("Failed to initialize ADS.");
    while (1);
  }}

void loop(void)
{
  int16_t results;

  /* Be sure to update this value based on the IC and the gain settings! */
  float   multiplier = 3.0F;    /* ADS1015 @ +/- 6.144V gain (12-bit results) */
  //float multiplier = 0.1875F; /* ADS1115  @ +/- 6.144V gain (16-bit results) */

  results = ads.readADC_Differential_0_1();

  Serial.print("Differential: "); Serial.print(results); Serial.print("("); Serial.print(results * multiplier); Serial.println("mV)");

  delay(1000);
}`

**Reply by DeeEmm** [reply]
What logic level converter are you using? is it pin compatible with the board?. Perhaps this is your issue? I had to use jumper wires on my board to address this issue with the sparkfun level converter.

**Reply by danielerpl** [reply]
I am using this one for ADS1115 https://www.az-delivery.de/it/products/analog-digitalwandler-ads1115-mit-i2c-interface
for PCA9306 I am using this https://www.amazon.it/dp/B0B778Z6RC/?coliid=I3OQLT9LCLXX1I&colid=1R5JBBXGKZGZX&psc=1&ref_=list_c_wl_lv_ov_lig_dp_it

Let me know if I can do something to help you to understand what problem I have 
Many thanks

**Reply by DeeEmm** [reply]
Just noticed that the test code you used above is incomplete. it is also set for ADS1015 not ADS1115. It does not show the I2C address or initialisation. You need to find out what the I2C address is for the ADS1115 unit that you have. The I2C address differs between manufacturers. If the address is incorrect, the I2C bus will not initialise.

Same thing applies to the other I2C devices on the bus. 

The I2C addresses are set in configuration.h

With regards to the logic level converter. As I mentioned above, I used the Sparkfun PCA9306. I would suggest comparing the pin connections on the DIYFB board, against the PCA9306 you have and make sure that it is pin compatible. It may be different. The pins may be in different locations.

**Reply by danielerpl** [reply]
Pinout of the PCA9306 it's the same 

I made a change on the test code, used the same library from DIYFB project but nothing....I don't have any connection with the ADS 

`#include <Arduino.h>

#include <SPI.h>

const int ADC_I2C_ADDR = 0x48; 

#include <ADS1115_lite.h>
ADS1115_lite adc(ADC_I2C_ADDR);

void setup(void)
{
  Serial.begin(115200);

   if (!adc.testConnection()) {
    Serial.println("ADS1115 Connection failed");
		while(1); //Freeze
	} else {
		Serial.println("ADS1115 Initialised\n");
  }

  }

void loop(void)
{
 
}`

**Reply by danielerpl** [reply]
<img width="126" alt="Screenshot 2024-01-15 at 18 10 14" src="https://github.com/DeeEmm/DIY-Flow-Bench/assets/14088795/4a6643ac-f5d7-48ab-9e53-c5c3018eb23a">
<img width="520" alt="Screenshot 2024-01-15 at 18 10 27" src="https://github.com/DeeEmm/DIY-Flow-Bench/assets/14088795/cbf56bac-3369-4410-8077-258dddfda7a6">

as you can see my PCA is the same pinout of your PCA by sparkFun, right ?

**Reply by DeeEmm** [reply]
Perhaps but the pinout of the Sparkfun is not the same as the board. As I mentioned I would suggest comparing the pin connections on the DIYFB board, against the PCA9306 you have and make sure that it is pin compatible. It may be different. The pins may be in different locations. If it is the same as the Sparkfun then it is not the same as the DIYFB board.

**Reply by danielerpl** [reply]
Really ? I have seen the pin on the DIY board and it's like the same, please can you give me a scratch about the difference ? I am not very expert in this type of things, I am better in coding. 
But, I am pretty sure the name on the DIY board it's like the pin on the PCA board.....I mean....

**Comment by DeeEmm** [comment]
You will need to confirm the pin connections for your board. If it is different just connect SCL1 to SCL1, SDA1 to SDA1 etc using jumper cables. 

If your board is the same as the Sparkfun board that I have as you have stated, then the pin locations are not compatible with the DIYFB PCB and you must connect by jumper cables as the pin locations are not the same.

You need to compare your board directly against the DIYFB PCB. it is no good comparing it to the Sparkfun board as this is incorrect.

This should be straightforwards, but I would advise to double check against the data sheet for your device to be sure.

Note the 'Known Issues'  stated at the top of the V2 PCB discussion - https://github.com/DeeEmm/DIY-Flow-Bench/discussions/129#discussion-4892058

>Non-breaking issues
>Pinouts for Sparkfun PCA9306 I2C level converter on PCB does not match the Sparkfun BOB that I have. Having now checked the >Footprint that I used, I can see that I did use what is supposed to be the Sparkfun footprint, however it is different from the physical >board I have. Not sure if this is a version thing or perhaps an error from Sparkfun. The issue is easily overcome by using PCB Jumper >Cables. Whilst not ideal, at least it works. I will review and update for V2.1 as the ideal scenario is that all of the BOBs are plug and play

**Reply by danielerpl** [reply]
So, I have the DIY board 2.0 and the print on the board it's like the print of the pin in my PCA board. What I got from you is...the pinout it's not the same like reported on the board. 

The schema of my PCA it's like the schema of the DIY pinout, I am not understanding with pin is different
Sorry for this confusion to explain 


<img width="724" alt="Screenshot 2024-01-19 at 15 33 29" src="https://github.com/DeeEmm/DIY-Flow-Bench/assets/14088795/8e8f5fac-9ec3-4725-b040-5f19fc17b198">


<img width="1149" alt="Screenshot 2024-01-19 at 15 35 21" src="https://github.com/DeeEmm/DIY-Flow-Bench/assets/14088795/00707a8c-c279-4f6d-b985-8f74aac50fb4">

**Reply by DeeEmm** [reply]
Before you said that your pins are the same as the Sparkfun board. 

The Sparkfun board is not the same as the PCB and must be connected by jumper leads

If your pins are the same as the PCB then you should be okay to plug in directly.

Also. You have still not confirmed the address of the I2C devices that you have are the same as the default addresses used in the code. You need to do this.

**Reply by DeeEmm** [reply]
BTW you need to check the physical screen print on the board, not the schematic - the schematic pin locations are not guaranteed to be the same as the physical pin locations - this was the exact issue I had with the Sparkfun board. The library part for the PCB design software was incorrect - no doubt as someone had created it from looking at the schematic and not the physical part

**Reply by danielerpl** [reply]
What I shared in the previous message it's the schema of my component near the image of the DIY board 2.0 
In this message I am sharing you the i2c address of the ADS1115 I mounted on the board 
It's the same of your, can you confirm ? 

I am sharing you also the manual of the component, it's in Italian, but you can see the address of the i2c

<img width="196" alt="Screenshot 2024-01-19 at 15 43 08" src="https://github.com/DeeEmm/DIY-Flow-Bench/assets/14088795/fd6186ac-2238-4397-886b-0a0339c41a24">


[Analog - Digital converter ADS1115 ITA.pdf](https://github.com/DeeEmm/DIY-Flow-Bench/files/13990639/Analog.-.Digital.converter.ADS1115.ITA.pdf)

**Reply by DeeEmm** [reply]
You can check the addresses in the configuration file  search for 'I2C_ADDR' - https://github.com/DeeEmm/DIY-Flow-Bench/blob/master/ESP32/DIY-Flow-Bench/configuration.h

**Reply by danielerpl** [reply]
const int ADC_I2C_ADDR = 0x48; 

there is this one

**Reply by DeeEmm** [reply]
You will also need to check the BME address as well.

**Reply by danielerpl** [reply]
I did a check for the BME and the address is 0x76, like in the configuration.h 
I also tried to change it in 0x77, but I am still receiving the error 

```
Found 0 device(s). 
Initialising ADS1115 
ADS1115 Connection failed
```

**Reply by DeeEmm** [reply]
This is with Wemos D1 ESP32 device?

**Reply by danielerpl** [reply]
Correct, i found your same one and i am using this 
If you want i can share some info
I also bought the digital analizer, i would try to debug these sensors to se what is going on

**Reply by DeeEmm** [reply]
Ok so if the addresses are both okay, the same code is working with other installs so I suspect that it is not code related.

We can confirm this by eliminating the DIYFB code and running a simple I2C test code with the shield attached, which I recall you have also tried?

Next I would breadboard the BME, ADC and Level Converter to see if it works without the shield. If this works it would indicate an issue with the PCB or assembly.

It is just a process of elimination to try and isolate where the issue is.

**Reply by danielerpl** [reply]
Yes, I already tried a test code for I2c, the code it's posted in an early message. Do you think it's a valid code ? 
So, to breadboard the components I have to desolder them, I will do it.. should be not a problem

**Reply by DeeEmm** [reply]
I've not tested the code so cannot comment. There are plenty of generic I2C test programs available on the net. If you want to test each device individually, usually the manufacturer provides test code examples.

One thing I note looking at the schematics is that the ADDR pin of the ADS1115 in actually floating, it is not connected to anything. You can try tying the ADDR pin to GND to see if that makes a difference. It is conceivable that if the voltage at the ADDR pin is high enough, the address is not defaulting to 0x48 but has changed to a different value. I will address this in the next version.

The other consideration when testing is that the BME I2C device is 3.3v and the ADS1115 is 5v, this is why the level shifter is used. However, for the purposes of testing the I2C it is sufficient to use 3.3v VCC for the ADS1115 and ignore the level shifter. This will simplify your test. The ADS1115 will work fine on 3.3v for establishing and testing I2C comms but you will only be able to measure a maximum of 3.3v

**Reply by danielerpl** [reply]
Ok, I will try this test and I will give you the feedback as soon as possible
 
Just one thing, I am not very sure about how to do the second test for the 3.3Vcc to the ADS1115 
Sorry for that but I would  be sure to make the right test for you

**Reply by DeeEmm** [reply]
The SDA/SCL lines are 3.3v, so if you connect the BME / ADS directly to the ESP it will work fine. (Don't forget that you will need pull-up resistors on both lines).

However, normally the ADS is connected to the 5v supply line so that it can measure 5v analog signals. The issue here is that as the ADS is now operating at 5v, the SDA/SCL lines are also now at 5v, which is too high for the ESP, which is a 3.3v device.

So in between the ADS and the ESP you need to convert the SDA/SCL lines from 5v to 3.3v by installing the PCA9306. This device has both 3.3v and 5v supplies. (VREF1 & VREF2). The associated SCA/SCL lines are then connected to the ADS and ESP as appropriate. Don't forget that you will need to add pull-up resistors to the 5v SDA/SCL lines

You can follow this on the schematic if you get stuck, but essentially it is something like this....                                                                                                                                                                                                        

![PXL_20240124_230727740](https://github.com/DeeEmm/DIY-Flow-Bench/assets/3038710/1c998adc-5d3f-43b7-866a-c1b951caf942)

There is also an interesting article I found when I was checking which voltage to put on VREF 1 / 2 - https://community.ultimaker.com/topic/13123-discrepancy-between-pca9306-datasheet-from-ti-and-ulticontroller-rev21-schematic/ - It states that the device can become damaged if exposed to voltages over 5.5v, which can include transient voltages from events such as PSU surge spikes at power up.

Let me know how you get on

**Reply by danielerpl** [reply]
Ok, clear 

before I desolder components from the board I could test which voltage there is on it. How can I do this ? 
Just to avoid the not necessary desoldering from the DIY PCB

**Reply by danielerpl** [reply]
I removed each component from the DIY PCB and I made I first test on the breadboard 

#BME 
```
I2C Scanner
Scanning...
I2C device found at address 0x76  !
done
```
Now we know which one is the correct address of the BME 0x76

#ADS115 
```
Scanning...
I2C device found at address 0x48  !
done
```
Now we know which one is the correct address of the ADS1115 0x48

about the PCA9306 which test can I do on the breadboard ?

**Reply by DeeEmm** [reply]
You can connect the devices in accordance with the sketch above.

For clarity the devices are as follows:

- PCA = PCA9306
- ADS = ADS1115
- ESP32 = ESP32
- BME = BME280

This is the same connection schema as in the schematic diagram.

**Reply by danielerpl** [reply]
Right!! for the resistor how much ? should be enough 1k ?

**Reply by DeeEmm** [reply]
The values are 4K7.

For reference you can refer to the schematic diagram - https://github.com/DeeEmm/DIY-Flow-Bench/discussions/129

**Reply by danielerpl** [reply]
There the possibility to have a new board? Or otherwise the design and the project and i will print it by myself? 
i think there is a problem in my board because each component individually it works very well

**Reply by DeeEmm** [reply]
So the test worked when connected individually but not on the board? It would point to there being an issue with the board.

Sorry to hear that.

I'm happy to send out a new board but I have not been able to get back to this project to get more boards made up. The design files have been revised and there is a small amount of work still to do before I can get a new batch made. But unfortunately at the moment I am tied up with house renovations so all projects like this have had to take a back seat. 

At the moment I do not have a date on when I will be able to get back to it.

I might suggest trying to track the issue on the board. It will be either a short or open circuit on the I2C bus. you can make a temporary repair by cutting the track if it is shorted or adding jumpers if it is O/C

The only other suggestion that I can make is to breadboard from the schematics, but take note of the known issues in the official PCB thread.

**Comment by danielerpl** [comment]
Hi Mick
many thanks for your response! All you are doing it's very appreciated and
I will never stop to thank you
I will try to fix the issue with a friend expert in electronics.
Just to help me in this process, can you share me the schema of the board ?
just to have the reference to make all check on it

Many thanks for your help! I hope you will be back in the project soon! I
hope I will fix soon, to make an help and improvements in your project!

Daniele

**Comment by danielerpl** [comment]
@DeeEmm I finally solved the problem for the board. The problem was the PCA9306. I removed it and made a direct wire from SDA1 to SDA2 
VREF1 to VREF2 
It works perfectly now! 
I also connected the board to my bench and I can see some value. 
It's time now to config it and make the first real test. I have only to figure out why I do not have the CFM value. I have the depressor but not the CFM. I have it but it's very low, around 0.9 
Any suggestion ?

**Reply by DeeEmm** [reply]
There is a minimum / pressure AND flow that needs to be met before the GUI will display a flow value. This means that if you measure only flow, it will show zero. You can adjust the threshold values under the Data filter section, in the configuration tab...

<img width="791" alt="Screenshot 2024-09-20 at 8 52 38 AM" src="https://github.com/user-attachments/assets/58130025-a95c-4fb7-a064-de158fc32a27">


The reason for using both flow and reference pressure values is that otherwise the GUI will display non-zero values at rest when the bench is not running. This is due to small quiescent voltage deviations in the system from the sensors, which when processed by the code result in a non-zero output. These small deviations are not enough to impact the flow readings in use, but without filtering means that the GUI flow value 'flickers'. So I added the thresholds to display a zero value when the bench is not running.

You can set these to zero for testing purposes.

**Reply by danielerpl** [reply]
ok clear, can I set your same values in the Data Filters section ?

**Comment by danielerpl** [comment]
@DeeEmm  about library dependencies I have to solve one little thing 

I have an error with the library ------ I can't solve the import and I have to comment this part of the code in the API.cpp 



`/***********************************************************
 * @name CREATE CRC32 CHECKSUM
 * @brief Compute checksum using ESP32 ROM CRC library
 * @returns CRC32 hash derived from *str
 * @implements <esp32/rom/crc.h>
 ***/
// uint32_t API::calcCRC (const char* str) {

//   CRC = (crc32_le(0, (const uint8_t*)str, 4));

//   return CRC;
// }`

How can I solve the dependencies ?

**Reply by DeeEmm** [reply]
The 'calcCRC' method is used to add a checksum to API calls so that data returned from API requests can be validated.

I am unsure why you are seeing a dependency issue as it compiles fine here and includes the call for the library file - #include <esp32/rom/crc.h>

If you are not utilising the API comms, then it is safe to disable the checksum by ensuring that "API_CHECKSUM_IS_ENABLED" is commented out in the configuration file. You can additionally comment out the function and any calls made to it.

I will take a look into library dependencies as I am wondering if I should actually be forking all libraries and then directing library calls to the forked versions to prevent future issues.

**Reply by danielerpl** [reply]
Ok perfect! I try again when you fork the library folder

**Comment by danielerpl** [comment]
@DeeEmm  i would like tomprint the last board design because i need a second board for testing at home before the upload to the proper one on the bench. Can you share the file to do it? I can also print the number of the boards for other people 
Let me know, we can arrange a common print order

**Reply by DeeEmm** [reply]
I have new PCBs here but I have not had time to validate them yet. i will let you know when this is done.
